### PR TITLE
feature: ENG-967 - Add dynamic app catalog

### PR DESCRIPTION
**Pull request by dogmar**
Pulls information from our API to show installation instructions for all apps in our catalog. Adding a page of the form `/pages/applications/[app-name].mdpart` will append extra information to the instructions from the API. If there are any headings in this markdoc file containing the word "configuration" (case-insensitive), the page will show that information in the markdoc file instead of the configuration from the API.

Make sure all pages in the `/pages/applications` directory end with `.mdpart`, otherwise the site will not render properly.

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = () => {
   const plugins = [withMarkdoc, withTM]
 
   return plugins.reduce((acc, next) => next(acc), {
-    reactStrictMode: true,
+    reactStrictMode: false,
```
Comment (dherault) [inline next.config.js:16]: Good, I like that. I love what strict mode puts on the table but I hate the double effects calls

File: package.json [hunk 2]
```diff
@@ -4,32 +4,37 @@
     "node": ">=18"
   },
   "scripts": {
-    "dev": "next dev",
+    "dev": "concurrently \"next dev\" \"graphql-codegen --watch --config codegen.ts \" ",
```
Comment (dherault) [inline package.json:7]: a && b --> a then b
a & b --> a and b concurrently
No need for a package

File: package.json [hunk 3]
```diff
@@ -4,32 +4,37 @@
...
+    "@react-aria/ssr": "3.4.0",
     "@react-stately/list": "3.5.4",
     "@react-stately/select": "3.3.2",
     "@tanstack/react-table": "8.5.28",
     "algoliasearch": "4.14.2",
     "classnames": "2.3.2",
     "deep-freeze": "0.0.1",
-    "honorable": "0.144.0",
+    "graphql": "16.6.0",
+    "honorable": "0.147.0",
```
Comment (dherault) [inline package.json:31]: You're like 40 versions behind ^^
Try:
`npm i -g npm-check-updates`
`ncu -i`

File: src/utils/text.ts [hunk 4]
```diff
@@ -6,18 +10,40 @@ export function removeTrailingSlashes(str) {
...
+
+export function isSubrouteOf(route: string, compareRoute:string) {
+  return route.startsWith(compareRoute)
+}
+
+export function isAppCatalogRoute(route: string) {
+  return isSubrouteOf(route, APP_CATALOG_BASE_URL)
+}
+
+export const providerToProviderName:Record<Provider, string> = {
```
Comment (dherault) [inline src/utils/text.ts:41]: Everything was perfect until the last declaration of the last file ^^
You missed a space here.

**Review by dherault — APPROVED** [review]
Mama sayscode a lot and well and I'll be proud.
My mama would be proud of you.

**Comment by dherault** [comment]
I'm noticing a weird behavior on the. deployed preview app:
`https://deploy-preview-102--pluraldocs.netlify.app/applications`  then click "Back/All docs" doesn't change the URL

**Comment by dherault** [comment]
Instead it adds an "App catalog" button that flickers a bitthe content

**Comment by dherault** [comment]
I say we remove that buttonand match the URL accordingly if that's possible easily.

File: src/data/getRepos.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+export const reposCache: {
+  filtered: RepoFragmentFragment[]
+} = {
+  filtered: [],
+}
+const filterRepos = memoizeOne((data: ReposQuery) => data?.repositories?.edges
+  ?.map(edge => edge?.node)
+  .map(node => asFragment(RepoFragment, node))
+  .filter((repo: RepoFragmentFragment | undefined | null): repo is RepoFragmentFragment => !!repo && !repo.private))
```
Comment (michaeljguarino) [inline src/data/getRepos.tsx:18]: I don't believe this query will ever return private repos, were you seeing that when unauthenticated?

File: src/NavData.tsx [hunk 6]
```diff
@@ -240,6 +266,28 @@ const data: NavData = [
...
+  repos,
+}: {
+  repos: FragmentType<typeof RepoFragment>[]
+}): NavData => ({
+  docs: rootNavData,
+  appCatalog: [
+    {
+      title: 'Application Catalog',
+      sections: [
+        { title: 'Repository Documentation', href: APP_CATALOG_BASE_URL },
```
Comment (avaidyanatha) [inline src/NavData.tsx:281]:
```suggestion
        { title: 'Catalog Overview', href: APP_CATALOG_BASE_URL },
```

File: src/data/getRepos.tsx (hunk 5 above)
Comment (dogmar) [reply src/data/getRepos.tsx]: Ah, I did almost all dev work while authenticated and and never changed this.

**Comment by dogmar** [comment]
> I say we remove that button and match the URL accordingly if that's feasible easily.

That's intended behavior. It's just supposed to navigate the menus, not actually change the page. Maybe it just needs different wording?

**Pull request merged**